Add reading of CSV files to ModelicaStandardTables

beutlich opened the pull request:
Taken from prototype implementation of [ModelicaTableAdditions](https://github.com/modelica-3rdparty/ModelicaTableAdditions). Note, that no additional external 3-rd party dependencies are utilized. CSV tables with quoted values or line breaks are not supported.

beutlich commented:
I update the binaries in my branch "update-binaries-add-csv". Use https://github.com/beutlich/ModelicaStandardLibrary/tree/update-binaries-add-csv if you want to test it.

dietmarw commented:
Could it be that you forgot to add some test models that are reading the CSV files that were added?

beutlich commented:
> 
> 
> Could it be that you forgot to add some test models that are reading the CSV files that were added?

Yes, I will add them, but wanted to gather some early feedback from library officers first.

MartinOtter commented:
Sorry, currently have no time to inspect this in detail.

HansOlsson commented on ModelicaTest/Resources/Data/Tables/test.csv:
```diff
@@ -0,0 +1,3 @@
+time, value
+0.0, 0.0	
+1,1
```
Why is there no end-of-line on the last line?
I see some problems with handling such files - and I certainly don't want to make that mandatory.

beutlich replied:
This was taken from the original test file provided by @dietmarw. Both cases, that is with and w/o trailing new-line are covered. If you prefer, I can add it here, too.

beutlich replied:
There are now also test models in ModelicaTest that refere to CSV files with trailing new-line. Please review again.

beutlich commented:
> 
> 
> I updated the binaries in my branch "update-binaries-add-csv". Use https://github.com/beutlich/ModelicaStandardLibrary/tree/update-binaries-add-csv if you want to test it.

@dietmarw I've just added the test models to ModelicaTest. If you want to test with Dymola 2021X make sure to set ``Advanced.PlaceDymolaSourceFirst = 2;` and use the provided binaries.

dietmarw reviewed (approved):
Approved. 
Note `Advanced.PlaceDymolaSourceFirst = 2;` was also necessary for Dymola 2020

HansOlsson reviewed (approved):
Seems good.
I tried rebuilding binaries and running test-case and the result matched the csv-file.
